### DISCUSSION TITLE
Tasks returning numpy array / Passing numpy arrays between tasks

### DISCUSSION THREAD

**Discussion by sakthi-tech4th** [flyteorg/flyte#2907] [Q&A]
I'm building a ML training workflow and my tasks return numpy arrays. When I try to run them, I get the following error message:
![image](https://user-images.githubusercontent.com/110414092/191523500-bc84b1c9-61ed-4be9-b699-c055ca55ed9e.png)

I also tried to override the 'allow_pickle' parameter to True. Then it started asking for the path
![image](https://user-images.githubusercontent.com/110414092/191538582-1c6d77ce-76c5-45e6-8a70-e77f4be130e4.png)

My command (since i'm testing) 
` python flyte_pipeline.py`
As the guide suggests.

My questions are:
1. Does Flyte support tasks returning Numpy arrays?
2. Does Flyte support passing numpy arrays between tasks? 
3. If I want to pass Numpy Arrays between the tasks, what is the best way to do it? Do you have any reference documentations?

Thanks and Regards,
Sakthi

**Comment by samhita-alla** [comment]
@sakthi-tech4th, NumPy array can be passed among the Flyte tasks. Yours is a NumPy array with `dtype` set to object and hence it's failing. It works with all the other `dtype`s. I'll add support to be able to serialize and deserialize object NumPy arrays when a user wants to pickle the data. Please bear with me while I work on the implementation.

**Reply by samhita-alla** [reply]
Created a PR: https://github.com/flyteorg/flytekit/pull/1209
